test/algorithms/regressors/test_neural_network_regressor.py

```diff
@@ -110,3 +121,9 @@ def test_regressor_with_opflow_qnn(self, config):
         # score the result
         score = regressor.score(self.X, self.y)
         self.assertGreater(score, 0.5)
+
+        # callback
+        if callback is not None:
+            self.assertTrue(all(isinstance(value, float) for value in history["values"]))
+            for weights in history["weights"]:
+                self.assertTrue(all(isinstance(weight, float) for weight in weights))
```

Comment:
same, check the number of weights.

Reply:
Added checks in https://github.com/Qiskit/qiskit-machine-learning/pull/151/commits/472ca8b0a20f9b860a098f6aeb8df7747bd2353e